examples/ddpg_pmsm_dq_current_control.py

```diff
@@ -20,83 +20,138 @@
 from gym.core import Wrapper
 from gym.spaces import Box, Tuple
 
+# This example shows how we can use GEM to train a reinforcement learning agent to control the current within
+# a permanent magnet synchronous motor three-phase drive
+# We use a deep-deterministic-policy-gradient (DDPG) agent
+# to determine which action must be taken on a continuous-control-set
+
 
 class AppendLastActionWrapper(Wrapper):
     """
     The following environment considers the dead time in the real-world motor control systems.
-    The real-world system changes its state, while the agent calculates the next action based on a previoulsly measured
-    observation. Therefore, for the agents it seems as if the applied action effects the state one step delayed.
-    (with a dead time of one time-step)
+    The real-world system changes its state, while the agent simultaneously calculates the next action based on a
+    previously measured observation.
+    Therefore, for the agents it seems as if the applied action affects the environment with one step delay
+    (with a dead time of one time step).
 
-    For complete observability of the system at each time-step we append the last played action of the agent to the
-    observation, because this action will be the one that is active in the next step.
+    For complete observability of the system at each time step we append the last selected action to the
+    observation, because this action will be the one that is active while the agent has to make the next decision.
     """
     def __init__(self, environment):
         super().__init__(environment)
+        # append the action space dimensions to the observation space dimensions
         self.observation_space = Tuple((Box(
             np.concatenate((environment.observation_space[0].low, environment.action_space.low)),
             np.concatenate((environment.observation_space[0].high, environment.action_space.high))
         ), environment.observation_space[1]))
 
     def step(self, action):
+
         (state, ref), rew, term, info = self.env.step(action)
+
+        # extend the output state by the selected action
         state = np.concatenate((state, action))
+
         return (state, ref), rew, term, info
 
     def reset(self, **kwargs):
+
         state, ref = self.env.reset()
+
+        # extend the output state by zeros after reset
+        # no action can be appended yet, but the dimension must fit
         state = np.concatenate((state, np.zeros(self.env.action_space.shape)))
+
         return state, ref
 
 
 if __name__ == '__main__':
+
     tf.compat.v1.disable_eager_execution()

```

Comment:
@atra94 
Why is it necessary to disable eager execution?
Is it only for debugging?

Reply:
I think @wkirgsn  introduced this once into another example and it has been used since then. Maybe he can answer this question. Did you try to run the script without this line?

Reply:
I can't recall what was the purpose, i would remove it if it is not necessary anymore.

Disabling eager execution would only speed things up in case it was really enabled.

Reply:
Removal did not trigger any errors. I dropped this line.